File: flask_api/app/dao/order_item_dao.py
from app.models.order_item import OrderItem
from app.models import db


class OrderItemDAO:

    @staticmethod
    def get_all_for_order(order_id):
        return OrderItem.query.filter_by(order_id=order_id).all()

    @staticmethod
    def get_by_id(order_item_id):
        return OrderItem.query.filter_by(id=order_item_id).first()

    @staticmethod
    def get_by_ids(order_item_ids):
        return OrderItem.query.filter(OrderItem.id.in_(order_item_ids)).all()

    @staticmethod
    def create(order_id, data):
        data["order_id"] = order_id
        oi = OrderItem(**data)
        db.session.add(oi)
        db.session.commit()
        return oi

    @staticmethod
    def update(order_item_id, data):
        oi = OrderItemDAO.get_by_id(order_item_id)
        if not oi:
            return None
        for key, value in data.items():
            if hasattr(oi, key):
                setattr(oi, key, value)
        db.session.commit()
        return oi

    @staticmethod
    def delete(order_item_id):
        oi = OrderItemDAO.get_by_id(order_item_id)
        if not oi:
            return False
        db.session.delete(oi)
        db.session.commit()
        return True

File: java_membership_api/src/main/java/com/example/membership_api/dao/OrderItemDAO.java
package com.example.membership_api.dao;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.example.membership_api.model.OrderItem;

import jakarta.persistence.EntityManager;
import jakarta.persistence.PersistenceContext;
import jakarta.transaction.Transactional;

@Repository
@Transactional
public class OrderItemDAO {

    @PersistenceContext
    private EntityManager entityManager;

    public List<OrderItem> getAllForOrder(String orderId) {
        String jpql = "SELECT oi FROM OrderItem oi WHERE oi.orderId = :orderId";
        return entityManager.createQuery(jpql, OrderItem.class)
                .setParameter("orderId", orderId)
                .getResultList();
    }

    public OrderItem getById(String orderItemId) {
        return entityManager.find(OrderItem.class, orderItemId);
    }

    public List<OrderItem> getByIds(List<String> orderItemIds) {
        String jpql = "SELECT oi FROM OrderItem oi WHERE oi.id IN :ids";
        return entityManager.createQuery(jpql, OrderItem.class)
                .setParameter("ids", orderItemIds)
                .getResultList();
    }

    public OrderItem create(OrderItem orderItem) {
        entityManager.persist(orderItem);
        entityManager.flush();  // Force insert to DB and get generated id
        return orderItem;
    }

    public OrderItem update(OrderItem orderItem) {
        return entityManager.merge(orderItem);
    }

    public boolean delete(String orderItemId) {
        OrderItem oi = getById(orderItemId);
        if (oi == null) {
            return false;
        }
        entityManager.remove(oi);
        return true;
    }
}
